src/screens/Followings/elements/RenderList/index.js

```diff
@@ -0,0 +1,157 @@
+    alignItems: 'center',
+    paddingVertical: 5
+  },
+  imageProfile: {
+    width: 48,
+    height: 48,
+    borderRadius: 48
+  },
+  wrapTextProfile: {
+    flex: 1,
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+    marginLeft: 12
+  },
+  textProfileUsername: {
+    fontFamily: fonts.inter[500],
+    fontWeight: 'bold',
+    fontSize: 14,
+    color: colors.black,
+    lineHeight: 16.94
+  },
+  textProfileFullName: {
+    fontSize: 12,
+    color: colors.gray,
+    fontFamily: fonts.inter[400],
+    lineHeight: 18,
+    flexWrap: 'wrap'
+  },
+  buttonFollowing: {
+    width: 88,
+    height: 36,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderColor: colors.bondi_blue,
+    borderWidth: 1,
+    borderRadius: 8
+  },
+  card: {
+    height: 68,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    width: '100%',
+    marginVertical: 10
+  }
+});
+
+const DomainList = (props) => {
+  const {item, isHashtag, handleSetFollow, handleSetUnFollow, onPressBody} = props;
+
+  return (
+    <View style={styles.card}>
+      <GlobalButton onPress={() => onPressBody(item)} buttonStyle={styles.wrapProfile}>
+        {!isHashtag ? (
+          <React.Fragment>
+            {item.image ? (
+              <Image
+                source={{
+                  uri: item.image
+                }}
+                style={styles.profilepicture}
+                width={48}
+                height={48}
+              />
+            ) : (
+              <View style={styles.profilepicture} />
+            )}
+          </React.Fragment>
+        ) : null}
+
+        <View style={styles.wrapTextProfile}>
+          <Text style={styles.textProfileUsername}>
+            {isHashtag && '#'}
+            {item.name}
+          </Text>
+          <Text style={styles.textProfileFullName} numberOfLines={1} ellipsizeMode={'tail'}>
+            {item.description ? item.description : ''}
+          </Text>
+        </View>
+      </GlobalButton>
+      {item.isunfollowed ? (
+        <GlobalButton onPress={handleSetFollow}>
+          <View style={styles.buttonFollow}>
+            <Text style={styles.textButtonFollow}>{isHashtag ? 'Join' : 'Follow'}</Text>
+          </View>
+        </GlobalButton>
+      ) : (
+        <GlobalButton onPress={handleSetUnFollow}>
+          <View style={styles.buttonFollowing}>
+            <Text style={styles.textButtonFollowing}>{isHashtag ? 'Joined' : 'Following'}</Text>
+          </View>
+        </GlobalButton>
+      )}
+    </View>
+  );
+};
+
+DomainList.propTypes = {
+  item: PropTypes.object,
+  isHashtag: PropTypes.bool,
+  onPressBody: PropTypes.func,
+  handleSetFollow: PropTypes.func,
+  handleSetUnFollow: PropTypes.func
+};
+
+DomainList.defaultProps = {
+  handleSetFollow: () => null,
+  handleSetUnFollow: () => null,
+  onPressBody: () => null
+};
+
+export default React.memo(DomainList);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall. However, there is a potential issue with the `DomainList` component's prop types. The `item` prop is defined as an object but there is no specification for its shape. This could lead to confusion or errors in the future if the expected properties of `item` are not clear. I suggest using `PropTypes.shape` to define the expected properties of `item`.

```diff
DomainList.propTypes = {
-  item: PropTypes.object,
+  item: PropTypes.shape({
+    name: PropTypes.string,
+    description: PropTypes.string,
+    image: PropTypes.string,
+    isunfollowed: PropTypes.bool
+  }),
  isHashtag: PropTypes.bool,
  onPressBody: PropTypes.func,
  handleSetFollow: PropTypes.func,
  handleSetUnFollow: PropTypes.func
};
```
This change will make it clearer what properties are expected in the `item` object and help prevent bugs related to incorrect or missing properties.


<!-- This is an auto-generated comment by OSS CodeRabbit -->